### ISSUE TITLE
`Flux.cpu` converts sparse cpu array to dense cpu array

### ISSUE THREAD

**Issue by CarloLucibello** [FluxML/Flux.jl#1720]
`x` is supposed to remain the same in the example below
```julia
using SparseArrays

julia> x = sprand(5,5,0.2)
5×5 SparseArrays.SparseMatrixCSC{Float64, Int64} with 3 stored entries:
  ⋅   0.881484   ⋅         ⋅    ⋅ 
  ⋅    ⋅        0.496493   ⋅    ⋅ 
  ⋅    ⋅         ⋅         ⋅    ⋅ 
  ⋅    ⋅         ⋅         ⋅    ⋅ 
  ⋅    ⋅         ⋅         ⋅   0.96554

julia> x |> cpu
5×5 Matrix{Float64}:
 0.0  0.881484  0.0       0.0  0.0
 0.0  0.0       0.496493  0.0  0.0
 0.0  0.0       0.0       0.0  0.0
 0.0  0.0       0.0       0.0  0.0
 0.0  0.0       0.0       0.0  0.96554
```

**Comment by DhairyaLGandhi** [comment]
Is this fixed on master?

**Comment by CarloLucibello** [comment]
no. This should fix it
```julia
_cpu_array(x::AbstractSparseMatrix) = adapt(SparseMatrixCSC, x)
```
I will file a PR

**Comment by DhairyaLGandhi** [comment]
closed in #1708
